analytics/src/analytics/integrations/etldb/main.py

```diff
@@ -190,3 +202,26 @@ def sync_quads(db: EtlDb, dataset: EtlDataset) -> dict:
                 f"QUAD '{ghid}' title = '{quad_df['quad_name']}', row_id = {result[ghid]}",
             )
     return result
+
+
+def get_sql_file_paths() -> dict[int, str]:
+    """Get all sql files needed for database initialization."""
+    result = {}
+
+    # define the path to the sql files
+    sql_file_directory = f"{Path(__file__).resolve().parent}/migrations/versions"
+
+    # get list of sorted filenames
+    filename_list = sorted(os.listdir(sql_file_directory))
+
+    # execute file only if filename follows convention: 0000_summary_of_change.sql
+    pattern = re.compile(r"^\d\d\d\d_.+\.sql$")
+    for filename in filename_list:
+        # validate filename format
+        if pattern.match(filename):
+            version = int(filename[:4])
+            result[version] = f"{sql_file_directory}/{filename}"
```

Comment:
Wanted to pull [this comment](https://github.com/HHS/simpler-grants-gov/pull/2870#discussion_r1846801062) to a top-level review thread since it was getting lost earlier in the PR.

If I'm understanding this correctly, it looks like if there are two SQL files with the same version, we'll end up skipping the first one (by overwriting its path in the map with the second file)

As @mdragon suggested, I'd strongly recommend that we fail and halt execution under two likely scenarios:
- Two files with the same version prefix
- A file that doesn't match this pattern

The current approach risks downstream problems by having certain migrations be silently skipped unless verbose is enabled.

Reply:
Done - please see latest commits